cmd/auparse/auparse.go

```diff
@@ -163,52 +165,52 @@ func outputMultipleMessages(msgs []*auparse.AuditMessage) {
 
 	switch *format {
 	case "json":
-		if err := printJSON(event); err != nil {
+		if err := s.printJSON(event); err != nil {
 			log.WithError(err).Error("failed to marshal event to JSON")
 		}
 	case "yaml":
-		fmt.Println("---")
-		if err := printYAML(event); err != nil {
+		s.output.Write([]byte("---\n"))
+		if err := s.printYAML(event); err != nil {
 			log.WithError(err).Error("failed to marshal message to YAML")
 		}
 	default:
 		sm := event.Summary
-		fmt.Println("---")
-		fmt.Printf(`time="%v" sequence=%v category=%v type=%v actor=%v/%v action=%v thing=%v/%v how=%v tags=%v`+"\n",
+		s.output.Write([]byte("---\n"))
+		s.output.Write([]byte(fmt.Sprintf(`time="%v" sequence=%v category=%v type=%v actor=%v/%v action=%v thing=%v/%v how=%v tags=%v`+"\n",
```

Comment:
In the places where you pair `io.Writer` with `fmt.SprintX` you can use `fmt.FprintX(io.Writer, ...)` which will simplify the code.